### PR TITLE
task-2

### PR DESCRIPTION
**Pull request by rameezitgithub**

### CODE REVIEW

File: app/Http/Controllers/TaskController.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public function index() {
+        $tasks = Task::orderBy('created_at', 'asc')->get();
+
+        return view('tasks', [
+            'tasks' => $tasks
+        ]);
+    }
+
+    public function delete(){
+        function ($id) {
```
Comment (AsfandYarAhmad) [inline app/Http/Controllers/TaskController.php]: remove `function ($id)` and pass id like `public function delete($id){`

File: app/Http/Controllers/TaskController.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Task;
+use Dotenv\Validator;
+use Illuminate\Auth\Events\Validated;
+use Illuminate\Contracts\Validation\Validator as ValidationValidator;
```
Comment (AsfandYarAhmad) [inline app/Http/Controllers/TaskController.php]: remove all the extra `Validator` class and import right class here

File: routes/web.php [hunk 3]
```diff
@@ -20,43 +21,8 @@
...
-            ->withErrors($validator);
-    }
-
-    $task = new Task;
-    $task->name = $request->name;
-    $task->save();
-
-    return redirect('/');
-});
+Route::post('/task', [TaskController::class,'validat'])->name('validat');
```
Comment (AsfandYarAhmad) [inline routes/web.php]: the function name is not right it should be like `update`

File: app/Http/Controllers/TaskController.php (hunk 1 above)
Comment (rameezitgithub) [reply app/Http/Controllers/TaskController.php]: problem solved

File: routes/web.php (hunk 3 above)
Comment (rameezitgithub) [reply routes/web.php]: problem solved

File: app/Http/Controllers/TaskController.php (hunk 2 above)
Comment (rameezitgithub) [reply app/Http/Controllers/TaskController.php]: solved

File: app/Http/Controllers/TaskController.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+            return redirect('/');
+        
+    }
+    public function  update(Request $request) {
+
+         
+            $validator = FacadesValidator::make($request->all(), [
+                'name' => 'required|max:20',
+            'name' => 'required|min:2',
```
Comment (AsfandYarAhmad) [inline app/Http/Controllers/TaskController.php]: why using validation on name two time 
remove line 29 and move `min:2` to line 28

File: app/Http/Controllers/TaskController.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            ]);
+
+            if ($validator->fails()) {
+                return redirect('/')
+                    ->withInput()
+                    ->withErrors($validator);
+            }
+        // $task = new Task;
+        // $task->name = $request->name;
+        // $task->save();
```
Comment (AsfandYarAhmad) [inline app/Http/Controllers/TaskController.php]: remove commented code

File: app/Http/Controllers/TaskController.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        return view('tasks', [
+            'tasks' => Task::store($request)
+        ]);
```
Comment (AsfandYarAhmad) [inline app/Http/Controllers/TaskController.php:30]: what is this? why are you passing the view store function

File: app/Http/Controllers/TaskController.php (hunk 4 above)
Comment (rameezitgithub) [reply app/Http/Controllers/TaskController.php]: solved

File: app/Http/Controllers/TaskController.php (hunk 5 above)
Comment (rameezitgithub) [reply app/Http/Controllers/TaskController.php]: solved

**Pull request merged**